p2p/upgrade.go

```diff
@@ -91,6 +98,7 @@ type Host struct {
 	}
 
 	host.Host
+	peerinfo.PeerInfo
 	pubsub.PubSub
```

Comment:
Do these all have to be public?

```suggestion
	host  host.Host
	pi     peerinfo.PeerInfo
	pubsub pubsub.PubSub
```

Reply:
Yes, it is important for these to be public b/c the methods in these interfaces are called on the `Host` (e.g. Host is passed to `activation.NewBuilder` to publish ATXs). We could have them as "plain members" and add accessors, but that will somewhat add to the code bloat

Reply:
On the second thought, refactored to have `PeerInfo()` method, but it's harder for `host.Host` and `pubsub.PubSub`

Reply:
The decising point on if the `Host` wraps or embeds those fields should be their relation ship. `Is` `Host` a `host.Host`? Probably yes - so embedding is OK. `Is` `Host` a `peerinfo.PeerInfo`? Probably not - it has peer info so it should not be embedded.

The other thing is as I said, having it embedded makes them public to all users of `Host` which might not always be intended, wrapping them allows to only expose some fields / methods of them as desired 🙂

Reply:
`PeerInfo` might be a bit of misnomer. It's more like `PeerInfoProvider` but that doesn't sound too good either. Will think a bit how to name it. But still `Host` being `PeerInfo` is probably not too good as `Host` is already a "God struct" with a lot of methods

Reply:
... for now probably we could leave it `PeerInfo` though, as `PeerInfoProvider` smells of Java DI, and not sure what a better name be ;)